tests/mesh/test-cartesian-grid.cpp

```diff
@@ -131,6 +131,165 @@ void test_cartesian_grids()
+    for( auto i : range( cartesiangrid.nb_cells_axis( 0 ) ) )
+    {
+        for( auto j : range( cartesiangrid.nb_cells_axis( 1 ) ) )
+        {
+            for( auto k : range( cartesiangrid.nb_cells_axis( 2 ) ) )
+            {
+                sivec3 position{ (signed_index_t) i, (signed_index_t) j,
+                    (signed_index_t) k };
+                index_t cell_offset = cartesiangrid.cell_offset( position );
+                if( f_indices_pointer->operator[]( cell_offset )
+                        != static_cast< float >( cell_offset )
+                    || u_indices_pointer->operator[]( cell_offset )
+                           != cell_offset
+                    || i_indices_pointer->operator[]( cell_offset )
+                           != static_cast< int >( cell_offset )
+                    || d_indices_pointer->operator[]( cell_offset )
+                           != static_cast< double >( cell_offset ) )
+                {
+                    throw RINGMeshException(
+                        "Test", "Error in attribute value #3." );
+                }
+            }
+        }
+    }
+    delete f_indices_pointer;
+    delete u_indices_pointer;
+    delete i_indices_pointer;
+    delete d_indices_pointer;
+
+    CartesianGridBuilder3D cgbuilder{ cartesiangrid };
+    cgbuilder.remove_section_from_cartesian_grid( 2, 4 );
+    if( cartesiangrid.nb_cells_axis( 2 ) != 8u )
+    {
+        throw RINGMeshException( "Test remove section",
+            "Wrong number of sections left : ",
+            cartesiangrid.nb_cells_axis( 2 ), "instead of 8." );
+    }
+    for( auto i : range( cartesiangrid.nb_cells_axis( 0 ) ) )
+    {
+        for( auto j : range( cartesiangrid.nb_cells_axis( 1 ) ) )
+        {
+            for( auto k : range( 3 ) )
+            {
+                sivec3 position{ (signed_index_t) i, (signed_index_t) j,
+                    (signed_index_t) k };
+                index_t cell_offset = cartesiangrid.cell_offset( position );
+                if( uindices[cell_offset] != cell_offset )
+                {
+                    throw RINGMeshException(
+                        "Test remove section", "Error in attribute value #1." );
+                }
+            }
+        }
+    }
+    for( auto i : range( cartesiangrid.nb_cells_axis( 0 ) ) )
+    {
+        for( auto j : range( cartesiangrid.nb_cells_axis( 1 ) ) )
+        {
+            for( auto k : range( 4, cartesiangrid.nb_cells_axis( 2 ) ) )
+            {
+                sivec3 position{ (signed_index_t) i, (signed_index_t) j,
+                    (signed_index_t) k };
+                index_t cell_offset = cartesiangrid.cell_offset( position );
+                if( uindices[cell_offset] != ( cell_offset + 10 * 8 ) )
+                {
+                    throw RINGMeshException(
+                        "Test remove section", "Error in attribute value #2." );
+                }
+            }
+        }
+    }
```

Comment:
Do you test cases where given id is out of range, when the type of attribute is not supported, when the size of the given attribute value vector is not the same of the grid? The objective is to try that the exceptions are correctly thrown.